src/components/Icon/Icon.tsx

```diff
@@ -130,10 +136,26 @@ class Icon extends UIComponent<Extendable<IIconProps>, any> {
     )
   }
 
+  isFontBased(themeIcons: ThemeIcons): boolean {
+    const { name } = this.props
+    const themeIcon = themeIcons[name]
+
+    return !themeIcon || typeof themeIcon !== 'function'
+  }
+
+  getExtraProps({ icons }): IconExtraProps {
+    const isFontBased = this.isFontBased(icons)
+
+    return {
+      isFontBased,
+      fontIconFromTheme: isFontBased && icons[this.props.name],
+    }
+  }
+
   renderComponent({ ElementType, classes, rest, accessibility, icons }) {
-    return this.props.svg
-      ? this.renderSvgIcon(ElementType, icons, classes, rest, accessibility)
-      : this.renderFontIcon(ElementType, classes, rest, accessibility)
+    return this.isFontBased(icons)
```

Comment:
I don't like the fact that you do similar calculation to what you've already done in `getExtraProps()`. Primarily not because of the performance but because for me (as a component developer) it is difficult to understand what the `extraProps` are good for - they are used to calculate styles, accessibility... why there are not used to render the component?
`renderComponent()` should receive `extraProps` as a parameter.

Reply:
Agreed, we can remove `getExtraProps` entirely if we check the type of the icon instead.  To limit the number of ideas and patterns in the library, I propose we also allow values or functions (as with variables and styles):

```jsx
const icons = {
  // value form
  umbrella: <svg></svg>,                             
  user: { fontFamily: 'Icons', content: "\\f0152" },

  // function form
  umbrella: () => <svg></svg>,
  user: () => ({ fontFamily: 'Icons', content: "\\f0152" }),
}

React.isValidElement(callable(icons.umbrella)()) ? isSVG : isFONT
```

This would allow us to pass values to the icon later.  It also removes the need for the `getExtraProps` function.